### DISCUSSION TITLE
System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Users\<User>\.wix\extensions\WixToolset.Util.wixext'.

### DISCUSSION THREAD

**Discussion by jweaver66** [oleg-shilo/wixsharp#1688] [Q&A]
I'm new to Wixsharp and I'm hoping you guys can help me. I am building a project to install a Windows Service and WPF application and need to have elevated permissions. 

When I execute my .msi via an elevated command prompt I still receive this message and cannot continue.
![image](https://github.com/user-attachments/assets/928eddc8-395b-40a0-88f3-06351d28811a)

Also, when I include any of these members of the ServiceInstaller I receive the exception in the title.
                                        FirstFailureActionType = FailureActionType.restart,
                                        SecondFailureActionType = FailureActionType.restart,
                                        ThirdFailureActionType = FailureActionType.runCommand,
                                        RestartServiceDelayInSeconds = 30,

My code: 
`  var project = new ManagedProject("foo",
                             new Dir(@"%ProgramFiles%\foo\UserInterface",
                                 new File(@"%ProgramFiles%\foo\foo.Sync.UI.exe")),
                              new Dir(@"%ProgramFiles%\foo\Service",
                                new File(@"%ProgramFiles%\foo\foo.Sync.Service.exe")
                                {
                                    ServiceInstaller = new ServiceInstaller("foo.Sync.Service.exe")
                                    {
                                        StartOn = SvcEvent.Install, //set it to null if you don't want service to start as during deployment
                                        StopOn = SvcEvent.InstallUninstall_Wait,
                                        RemoveOn = SvcEvent.Uninstall_Wait,
                                        Start = SvcStartType.auto
                                    }
                                }
                              ));

            project.GUID = new Guid("ba481d8b-f6ad-451e-a800-cacc5223f8e3");

            project.ManagedUI = new ManagedUI();
            project.ManagedUI.InstallDialogs.Add<WelcomeDialog>()
                                            .Add<LicenceDialog>()
                                            .Add<ProgressDialog>()
                                            .Add<ExitDialog>();

            project.ManagedUI.ModifyDialogs.Add<MaintenanceTypeDialog>()
                                           .Add<FeaturesDialog>()
                                           .Add<ProgressDialog>()
                                           .Add<ExitDialog>();

           Compiler.BuildMsi(project, $"setup.v{project.Version}.msi");  `

**Comment by oleg-shilo** [comment]
The exception means that the required wix extension is not found on your system.
In such situations WixSharp tries to install the extension on fly but most likely failed. Lately it was a spike with WiX extensions not [being installed correctly](https://github.com/oleg-shilo/wixsharp/issues/1671#issuecomment-2495720605). It looks like the deployment mechanism is not mature enough. WiX even recognizes and marks some corrupted deployments as damaged:

![image](https://github.com/user-attachments/assets/211ffff4-a5c0-4637-ac30-d83a15c72671)

The link to the issue above may help. But to put is simple, you have to install the extension manually. Unfortunately.

```ps1
wix extension add -g WixToolset.Util
```

**Reply by maroonedsia** [reply]
Hi Oleg, I still have the same problem. I'm using Wix `6.0.2+b3f3403` and `"WixSharp_wix4" Version="2.6.2"`. Running `wix extension add -g WixToolset.Util` looks to do nothing, and running `wix extension list` after that doesn't show any installed extensions. Copilot's suggestion was to reference this nuget package `"WixToolset.UI.wixext" Version="6.0.2"` which is not resolving the build error, neither. 

Any idea, please? 😎

**Reply by maroonedsia** [reply]
I could get unblocked by this workaround: 
Manually create the "extensions\WixToolset.UI.wixext" subfolders, then copy the content of `C:\Users\<user-name>\.nuget\packages\wixtoolset.ui.wixext\6.0.2` to `C:\Users\<user-name>\.wix\extensions\WixToolset.UI.wixext`.

**Reply by oleg-shilo** [reply]
Great, thank you for sharing the workaround.
The problem you are facing is caused by both WiX package managers (wix.exe and nuget) failing to bring the required extension to the local system. It's not much WixSharp can do about this. 

Your workaround fully confirms this interpretation as the user manually does what wix.exe is supposed to do. 

I just committed the fix #1851 that may actually fix the NuGet problem though.
